priv/templates/phx.gen.auth/schema.ex

```diff
@@ -116,5 +130,46 @@ defmodule <%= inspect schema.module %> do
     else
       add_error(changeset, :current_password, "is not valid")
     end
-  end
+  end<%= if requires_inspect_impls? do %>
+
+  ###
+  # This `Inspect` implementation is only needed for Elixir versions < 1.8
+  # and prevents `inspect/2` from returning sensitive fields.
+  #
+  # This can be removed after upgrading to Elixir 1.8+ and sensitive
+  # fields can be controlled with the the following line at the top of this
+  # module.
+  #
+  #     @derive {Inspect, except: [:password]}
+  ###
+  if Version.compare(System.version(), "1.8.0") == :lt do
+    defimpl Inspect do
+      @fields_to_exclude [:password]
+
+      import Inspect.Algebra
+
+      def inspect(<%= schema.singular %>, opts) do
+        colorless_opts = %{opts | syntax_colors: []}
+        name = Inspect.Atom.inspect(@for, colorless_opts)
+
+        open = color("#" <> name <> "<", :map, opts)
+        sep = color(",", :map, opts)
+        close = color(">", :map, opts)
+
+        map = Map.drop(<%= schema.singular %>, @fields_to_exclude ++ [:__struct__, :__exception__])
+
+        # Use the :limit option and an extra element to force
+        # `container_doc/6` to append "...".
+        opts = %{opts | limit: min(opts.limit, map_size(map))}
+
+        field_list =
+          map
+          |> Map.to_list()
+          |> Kernel.++(["..."])
+
+        container_doc(open, field_list, close, opts, &Inspect.List.keyword/2, separator: sep, break: :strict)
+
+      end
+    end
+  end<% end %>
```

Comment:
@josevalim In order to get the `inspect/2` test to pass on Elixir 1.7.4, I add a custom inspect implementation if the code is generated with Elixir 1.7. I also added an if statement that checks the elixir version in the generated code, so the user will automatically be transitioned to use the `@derive` directive when they upgrade to Elixir 1.8. Finally, I added some comments indicating what code can be removed when they upgrade.

Is this a good approach for handling this issue? It seems like a lot of extra code, but since phoenix targets 1.7+ by default, I thought it wouldn't be good to let the passwords be inspected on 1.7 while they're filtered on 1.8+.

Reply:
Since this will most likely be a separate package for now, we can make the package require v1.8+. By the time it lands on Phoenix, Phoenix will most likely require v1.8+ for new apps, so we should be good. In other words, I don't think we need to worry about this. :)

Reply:
Perfect! Thank you!